tests/MongoDB.EntityFrameworkCore.FunctionalTests/Update/ContextTests.cs

```diff
@@ -0,0 +1,223 @@
+
+        const int insertCount = 9;
+
+        db.Entitites.AddRange(
+            Enumerable.Range(0, insertCount).Select(i => new Customer("Generated " + i)));
+
+        Assert.Equal(insertCount, await db.SaveChangesAsync());
+        Assert.Equal(insertCount, await db.Entitites.CountAsync());
+    }
+
+    [Fact]
+    public void SaveChanges_includes_update_counts()
+    {
+        var db = SingleEntityDbContext.Create(tempDatabase.CreateTemporaryCollection<Customer>());
+
+        const int updateCount = 8;
+        var items = Enumerable.Range(0, updateCount * 2)
+            .Select(i => new Customer("Generated " + i))
+            .ToArray();
+
+        db.Entitites.AddRange(items);
+        db.SaveChanges();
+
+        for (int i = 0; i < updateCount; i++)
+            items[i].name = "Updated " + i;
+
+        Assert.Equal(updateCount, db.SaveChanges());
+        Assert.Equal(updateCount, db.Entitites.Count(c => c.name.StartsWith("Updated ")));
+    }
+
+    [Fact]
+    public async Task SaveChangesAsync_includes_update_counts()
+    {
+        var db = SingleEntityDbContext.Create(tempDatabase.CreateTemporaryCollection<Customer>());
+
+        const int updateCount = 7;
+        var items = Enumerable.Range(0, updateCount * 2)
+            .Select(i => new Customer("Generated " + i))
+            .ToArray();
+
+        db.Entitites.AddRange(items);
+        await db.SaveChangesAsync();
+
+        for (int i = 0; i < updateCount; i++)
+            items[i].name = "Updated " + i;
+
+        Assert.Equal(updateCount, await db.SaveChangesAsync());
+        Assert.Equal(updateCount, await db.Entitites.CountAsync(c => c.name.StartsWith("Updated ")));
+    }
+
+    [Fact]
+    public void SaveChanges_includes_delete_counts()
+    {
+        var db = SingleEntityDbContext.Create(tempDatabase.CreateTemporaryCollection<Customer>());
+
+        const int deleteCount = 6;
+        var items = Enumerable.Range(0, deleteCount * 2)
+            .Select(i => new Customer("Generated " + i))
+            .ToArray();
+
+        db.Entitites.AddRange(items);
+        db.SaveChanges();
+
+        db.Entitites.RemoveRange(items.Take(deleteCount));
+
+        Assert.Equal(deleteCount, db.SaveChanges());
+        Assert.Equal(deleteCount, db.Entitites.Count());
+    }
+
+    [Fact]
+    public async Task SaveChangesAsync_includes_delete_counts()
+    {
+        var db = SingleEntityDbContext.Create(tempDatabase.CreateTemporaryCollection<Customer>());
+
+        const int deleteCount = 6;
+        var items = Enumerable.Range(0, deleteCount * 2)
+            .Select(i => new Customer("Generated " + i))
+            .ToArray();
+
+        await db.Entitites.AddRangeAsync(items);
+        await db.SaveChangesAsync();
+
+        db.Entitites.RemoveRange(items.Take(deleteCount));
+
+        Assert.Equal(deleteCount, await db.SaveChangesAsync());
+        Assert.Equal(deleteCount, await db.Entitites.CountAsync());
```

Comment:
I would say we should expect `items.Count() - deleteCount` here instead, because we checking number of documents left in database after the deletion.